Add upgrading to 4.x docs (#6930)
nginxinc/kubernetes-ingress: site/content/upgrading-to-v4.md

---
description: Guide on upgrading to v4 and above
docs: DOCS-000

> title: Guild on Upgrading to v4 and above

Reviewer: ```suggestion
title: Guide on upgrading to NGINx Ingress Controller 4.x.x from 3.x.x
```

Reviewer 2: Do we need to say F5 NGINX Ingress Controller in this context?

Reviewer: @ADubhlaoich what do yo think?

Reviewer 3: No need to prefix the product name in page titles.

The hard rule is first mention on the body text of a page.

I'm in general agreement that the page title has space to improve.

toc: true
weight: 2100
headless: true
---

{{< note >}}

> If you are using NGINX Ingress Controller version `v3.3.2` (Helm Chart `v1.0.2`) or earlier, please upgrade to `v3.7.2` (Helm Chart `v1.4.2`), and convert your TransportServer and GlobalConfiguration resources to `apiVersion` `k8s.nginx.org/v1`.

Reviewer: ```suggestion
If you are using NGINX Ingress Controller version `v3.3.2` (Helm Chart `v1.0.2`) or earlier, please upgrade to `v3.7.2` (Helm Chart `v1.4.2`), and convert your TransportServer, GlobalConfiguration and Policy resources to `apiVersion` `k8s.nginx.org/v1`.
```

Author: Policy was at `apiVersion` `v1` much earlier than this. 4/5+ years ago from Git blame.

{{</ note >}}

> 1. If using NGINX Ingress Controller resources with an `apiVersion` of `k8s.nginx.org/v1alpha1`, prior to upgrading to `v4.x` (Helm Chart `v2.x`) , **please ensure** that they are updated to `apiVersion: k8s.nginx.org/v1`

Reviewer: ```suggestion
1. If using NGINX Ingress Controller custom resources with an `apiVersion` of `k8s.nginx.org/v1alpha1`, prior to upgrading to `v4.x` (Helm Chart `v2.x`) , **please ensure** that they are updated to `apiVersion: k8s.nginx.org/v1`
```
